pkgs/development/libraries/libclc/default.nix

```diff
@@ -1,33 +1,30 @@
-{ lib, stdenv, fetchFromGitHub, python, llvmPackages }:
+{ lib, stdenv, fetchFromGitHub, ninja, cmake, python, llvmPackages }:
 
 let
   llvm = llvmPackages.llvm;
   clang = llvmPackages.clang;
   clang-unwrapped = llvmPackages.clang-unwrapped;
 in
 
-stdenv.mkDerivation {
-  name = "libclc-2019-06-09";
+stdenv.mkDerivation rec {
+  name = "libclc";
+  version = "11.0.1";
 
   src = fetchFromGitHub {
-    owner = "llvm-mirror";
-    repo = "libclc";
-    rev = "9f6204ec04a8cadb6bef57caa71e3161c4f398f2";
-    sha256 = "03l9frx3iw3qdsb9rrscgzdwm6872gv6mkssvn027ndf9y321xk7";
+    owner = "llvm";
+    repo = "llvm-project";
+    rev = "llvmorg-${version}";
+    sha256 = "0bxh43hp1vl4axl3s9n2nb2ii8x1cbq98xz9c996f8rl5jy84ags";
   };
+  sourceRoot = "source/libclc";
 
-  nativeBuildInputs = [ python ];
-  buildInputs = [ llvm clang clang-unwrapped ];
-
+  # cmake expects all required binaries to be in the same place, so it will not be able to find clang without the patch without the patch
   postPatch = ''
-    sed -i 's,llvm_clang =.*,llvm_clang = "${clang-unwrapped}/bin/clang",' configure.py
-    sed -i 's,cxx_compiler =.*,cxx_compiler = "${clang}/bin/clang++",' configure.py
-  '';
-
-  configurePhase = ''
-    ${python.interpreter} ./configure.py --prefix=$out
+    sed -i 's,find_program( LLVM_CLANG clang PATHS ''${LLVM_BINDIR} NO_DEFAULT_PATH ),find_program( LLVM_CLANG clang PATHS "${clang-unwrapped}/bin" NO_DEFAULT_PATH ),' CMakeLists.txt
```

Comment:
Can we use a patch instead?

Reply:
Shure, will do

Reply:
Ah, I found the reason I didn't use a patch here: The cmake file searches for llvm clang in the same path as the other llvm stuff, but on Nix that is in different paths. Easiest solution was to replace the path where llvm clang is searched with `"${clang-unwrapped}/bin"`, sadly that is obviously dynamic.
I could patch the cmake file to accept an env var or similar, but that still feels meh.

Reply:
I think this sed is rather long and complex and we won't notice when it changes. We should probably replace it with a substituteInPlace or patch to know when it breaks.

Reply:
I didn't know about `substituteInPlace`, I will use that. The `sed` command sadly has to be that long, because `LLVM_BINDIR` is used a lot in the Makefile, but only in this case is it used to find `clang` and not `llvm*`.